lib/colord/cd-color.c

```diff
@@ -1465,12 +1467,17 @@ cd_color_get_blackbody_rgb_full (gdouble temp,
 	}
 
 	/* bilinear interpolate the blackbody data */
-	alpha = ((guint) temp % 100) / 100.0;
-	temp_index = ((guint) temp - 1000) / 100;
-	cd_color_rgb_interpolate (&blackbody_func[temp_index],
-				  &blackbody_func[temp_index + 1],
-				  alpha,
-				  result);
+	temp_quot = (guint) temp / 100;
+	temp_rem = (guint) temp % 100;
+
+	if (temp_rem == 0)
+		*result = blackbody_func[temp_quot - 10];
```

Comment:
I think it would be better to just return TRUE here and not indent the interpolate bit. Maybe also you can add a comment where the `- 10` comes from. Thanks.